Modules/modFileEncoding.ACM

```diff
@@ -0,0 +1,486 @@
+    lngsMap(140) = &H152:  lngsMap(141) = &H3F:   lngsMap(142) = &H17D:  lngsMap(143) = &H3F
+    lngsMap(144) = &H3F:   lngsMap(145) = &H2018: lngsMap(146) = &H2019: lngsMap(147) = &H201C
+    lngsMap(148) = &H201D: lngsMap(149) = &H2022: lngsMap(150) = &H2013: lngsMap(151) = &H2014
+    lngsMap(152) = &H2DC:  lngsMap(153) = &H2122: lngsMap(154) = &H161:  lngsMap(155) = &H203A
+    lngsMap(156) = &H153:  lngsMap(157) = &H3F:   lngsMap(158) = &H17E:  lngsMap(159) = &H178
+  End If
+  
+End Function
+
+Function ANSICP_to_UnicodeCP(bytANSI As Byte, ByRef bytUnicodeMSB As Byte, ByRef bytUnicodeLSB As Byte) As Boolean
+  ' Takes a single byte ANSI codepoint (character) and converts it to 16 bit Unicode placing the answer
+  ' in 2 separate Bytes representing the MSB (Most Significant Byte) and LSB (Least Significant Byte)
+  Static bytsUnicodeBestFit() As Long
+  Const lngLSBMask = 255 ' constant with Least Significant 8 bits set to 1
+  
+  ANSICP_to_UnicodeCP = False ' initially set to False
+  
+  If bytANSI < 128 Or bytANSI >= 160 Then
+    bytUnicodeMSB = 0
+    bytUnicodeLSB = bytANSI
+  Else
+    ' in this range we neede to use the best  fit lookup table
+    Call PopulateANSI_Unicode_BestFit(bytsUnicodeBestFit)
+    bytUnicodeMSB = Fix(bytsUnicodeBestFit(bytANSI) / 256)
+    bytUnicodeLSB = bytsUnicodeBestFit(bytANSI) And lngLSBMask
+  End If
+        
+  ANSICP_to_UnicodeCP = True
+  
+End Function
+
+Function UnicodeCP_to_ANSICP(bytUnicodeMSB As Byte, bytUnicodeLSB As Byte, ByRef bytANSI As Byte) As Boolean
+  ' Take a 2 byte ANSI codepoint and converts it to 8 bit ANSI placing the answer in the byt provided
+  ' MSB = Most Significant Byte, LSB = Least Significant Byte
+  ' Returns False if the character cannot be mapped
+  
+  Static bytsUnicodeBestFit() As Long
+  Dim lngUnicode As Long
+  Dim i As Long
+  
+  UnicodeCP_to_ANSICP = False ' intially set to False
+  
+  If bytUnicodeMSB = 0 Then
+    ' we have a single byte value
+    If bytUnicodeLSB < 128 Then
+      bytANSI = bytUnicodeLSB
+      UnicodeCP_to_ANSICP = True
+    ElseIf bytUnicodeLSB >= 160 Then
+      bytANSI = lngUnicode
+      UnicodeCP_to_ANSICP = True
+    'Else
+      ' we have a value in the range 128 (&H80) to 9F (159) which shouldn't be possible so do nothing to return a fail
+    End If
+  Else
+    ' we have a codepoint taking 2 bytes so see if it appears in our bestfit table
+    Call PopulateANSI_Unicode_BestFit(bytsUnicodeBestFit) ' populate the best fit table (if not already done)
+    lngUnicode = bytUnicodeMSB * 256 + bytUnicodeLSB ' calculate the full Unicode value
+    For i = 128 To 159
+      If bytsUnicodeBestFit(i) = lngUnicode Then
+        bytANSI = i
+        UnicodeCP_to_ANSICP = True
+        Exit For
+      End If
+    Next i
+  End If
+End Function
+
+' The 2 Functions UTF8CP_to_UCS2CP and UCS2CP_to_UTF8CP convert individual Codepoints (characters) to/from the 2 encodings

```

Comment:
Not to nitpick but a codepoint in not necessarily a character/character is a confusing word. See https://stackoverflow.com/a/27331885

```suggestion
' The 2 Functions UTF8CP_to_UCS2CP and UCS2CP_to_UTF8CP convert individual Codepoints to/from the 2 encodings
```

Reply:
Left as is.

Reply:
I agree that for our uses (primarily english) character and codepoint are interchangable, and `(character)` gives someone an idea of what a codepoint is.